src/groups/mqb/mqba/mqba_clientsession.cpp

```diff
@@ -2914,11 +3338,12 @@ void ClientSession::tearDown(const bsl::shared_ptr<void>& session,
     // NOTE: We use the e_DISPATCHER type because this Client is dying, and the
     //       process of this event by the dispatcher should not add it to the
     //       flush list.
-    dispatcher()->execute(bdlf::BindUtil::bind(&ClientSession::tearDownImpl,
-                                               this,
-                                               &semaphore,
-                                               session,
-                                               isBrokerShutdown),
+    dispatcher()->execute(bdlf::BindUtil::bindS(d_state.d_allocator_p,
+                                                &ClientSession::tearDownImpl,
+                                                this,
+                                                &semaphore,
+                                                session,
+                                                isBrokerShutdown),
```

Comment:
`tearDown` sequence is one place where we don't use `ClientSessionDestructGuard`, because `d_self_sp` is already invalidated. We cannot check it anymore in `ClientSessionDestructGuard`